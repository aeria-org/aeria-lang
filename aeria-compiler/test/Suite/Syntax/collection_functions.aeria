collection CollectionFunctionsTest {
  functions {
    customFunc1?
    customFunc2? @expose
    get @expose("unauthenticated")
    insert @expose(["root"])
    delete
  }
}
