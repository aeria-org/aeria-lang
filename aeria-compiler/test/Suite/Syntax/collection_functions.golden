{
  "kind": "Program",
  "collections": [
    {
      "name": {
        "value": "CollectionFunctionsTest",
        "kind": "CollectionName"
      },
      "kind": "Collection",
      "functions": [
        {
          "kind": "FunctionItem",
          "functioName": {
            "value": "customFunc1",
            "kind": "FunctionName"
          },
          "custom": true
        },
        {
          "kind": "FunctionItem",
          "functioName": {
            "value": "customFunc2",
            "kind": "FunctionName"
          },
          "expose": {
            "kind": "Attribute",
            "attributeValue": {
              "literal": {
                "value": true,
                "kind": "LBoolean"
              },
              "kind": "ALiteral"
            },
            "attributeName": {
              "value": "expose",
              "kind": "AttributeName"
            }
          },
          "custom": true
        },
        {
          "kind": "FunctionItem",
          "functioName": {
            "value": "get",
            "kind": "FunctionName"
          },
          "expose": {
            "kind": "Attribute",
            "attributeValue": {
              "literal": {
                "value": "unauthenticated",
                "kind": "LString"
              },
              "kind": "ALiteral"
            },
            "attributeName": {
              "value": "expose",
              "kind": "AttributeName"
            }
          },
          "custom": false
        },
        {
          "kind": "FunctionItem",
          "functioName": {
            "value": "insert",
            "kind": "FunctionName"
          },
          "expose": {
            "kind": "Attribute",
            "attributeValue": {
              "literal": {
                "value": [
                  {
                    "value": "root",
                    "kind": "LString"
                  }
                ],
                "kind": "LArray"
              },
              "kind": "ALiteral"
            },
            "attributeName": {
              "value": "expose",
              "kind": "AttributeName"
            }
          },
          "custom": false
        },
        {
          "kind": "FunctionItem",
          "functioName": {
            "value": "remove",
            "kind": "FunctionName"
          },
          "custom": false
        }
      ]
    }
  ]
}
